src/General/Alert/Alert.tsx

```diff
@@ -8,59 +8,40 @@ import {
   InfoSolidIcon,
   CloseIcon,
 } from '../Icon/components';
-import {
-  AlertContainer,
-  AlertContent,
-  AlertMessage,
-  AlertIcon,
-  AlertTypeIconContainer,
-} from './AlertStyle';
+import { AlertContainer, AlertIcon } from './AlertStyle';
 
 import { Greyscale, PrimaryColor, SecondaryColor } from '../../Utils/Colors';
+import { Box } from '../../Layout/Box';
+import { isUndefined } from 'lodash';
+import { DelayedUnmount } from '../../Utils/DelayedUnmount';
 
 export const Alert = ({
-  isOpen,
+  isOpen: externalIsVisible,
   autoClose,
-  onClose,
+  onClose: externalOnClose,
   type = 'info',
   className,
   message,
 }: Props) => {
-  const alertContainerRef = React.useRef<HTMLDivElement>();
-  const autoCloseTimeout = React.useRef<ReturnType<typeof setTimeout>>();
-  const [isVisible, setIsVisible] = React.useState<boolean>(isOpen);
+  const [internalIsVisible, setInternalIsVisible] = React.useState<boolean>(
+    true
+  );
 
-  React.useEffect(() => {
-    if (isOpen && !isVisible) {
-      setIsVisible(true);
-    }
-    if (!isOpen && isVisible) {
-      setTimeout(() => setIsVisible(false), 300);
-    }
-  }, [isOpen, isVisible, setIsVisible]);
```

Comment:
Simplified the logic so that there's only one var that determines whether the alert is open/visible or not.

Reply:
Hmm, what if the user passes the `isOpen={true}` and `auto={300}` but doesn't pass `onClose`. The Alert won't be closed right?

Reply:
Correct. I think that's reasonable behavior, because the alert would be doing exactly what the user is asking: Be open.

Reply:
Yes, it is reasonable behavior, but I was worried about this change is not backward-compatible which might break something. Have you checked that? 🙏

Reply:
There's only one usage on Employers and three on DST

![Screenshot from 2021-03-10 14-48-19](https://user-images.githubusercontent.com/3341028/110588922-a6f17000-81b0-11eb-8618-b7e763596943.png)
![Screenshot from 2021-03-10 14-48-09](https://user-images.githubusercontent.com/3341028/110588924-a8229d00-81b0-11eb-8b9f-4c6ed9172ca6.png)
![Screenshot from 2021-03-10 14-47-48](https://user-images.githubusercontent.com/3341028/110588928-a8bb3380-81b0-11eb-9506-522ee2d42139.png)
![Screenshot from 2021-03-10 14-46-58](https://user-images.githubusercontent.com/3341028/110588929-a953ca00-81b0-11eb-9ae7-945ce1449e8c.png)

Reply:
They all render the Alert unconditionally and control it's state. I don't know how to trigger those alerts so I did not check them, but I created my own in Employers like this and it works:

```
  const [open, setOpen] = useState(true);
  return (
    <Switch>
      <BaseRoute exact={true} path="/">
        <Alert
          type="danger"
          message="this sucks"
          isOpen={open}
          onClose={() => setOpen(false)}
          autoClose={7000}
        />
```

So clearly it is correct to have it call `onClose` when the specified timeout triggers.